addons/koji/common/src/main/java/org/commonjava/indy/koji/content/KojiMavenMetadataProvider.java

```diff
@@ -171,6 +171,8 @@ public Metadata getMetadata( StoreKey targetKey, String path )
             return null;
         }
 
+        checkKojiMDCapacity();
```

Comment:
Yeah, it worth to do that. BTW, i am aware that there are still some thread pools in indy, e.g.: the event dispatch executor, when i tried to add the check*Capacity there, a stack of classes need to be updated as well since it throws a new exception, it would be better to throw a runtime one. Thought ?

Reply:
A runtime exception might be good here. If we throw a checked exception in the interface, we have to handle them all over the world. I guess this is not necessary because not all pools are load sensitive, ie. some checks some does not. I am thinking we should use @ExecutorConfig to manage it. If the developer thinks a pool is load-sensitive, he should go handle the runtime exception. The default is false for other pools. To be flexible, the threadpool.conf should be able to alter a specific pool's setting but we don't need a global flag. I came up with a PR pls review it. 
Sorry, I did not tell you guys beforehand just because this is easy - the code is almost there when I finish thinking about it.